test/e2e/cluster.go

```diff
@@ -167,3 +180,42 @@ func createLoadBalancerService(ctx context.Context, cli kubernetes.Interface, na
 	_, err := cli.CoreV1().Services(testNamespace).Create(context.Background(), svc, metav1.CreateOptions{})
 	return err
 }
+
+func createContainerFromInternalContainerRegistryImage(ctx context.Context, cli kubernetes.Interface, name string) error {
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: testNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": name,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": name},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "cli",
+							Image: "image-registry.openshift-image-registry.svc:5000/openshift-image-registry/cli",
+							Command: []string{
+								"/bin/sh",
+								"-c",
+								"while true; do sleep 1; done",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err := cli.AppsV1().Deployments(testNamespace).Create(context.Background(), deploy, metav1.CreateOptions{})
+	return err
+}
+
+func int32Ptr(i int32) *int32 { return &i }
```

Comment:
There's a package `"github.com/Azure/go-autorest/autorest/to"` which provides a function similar to this which we use in our code already. Since we're in a test there isn't a lot of overhead for importing additional packages so I would use that.